src/test/test_group_np_cli.c

```diff
@@ -0,0 +1,178 @@
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted for any purpose (including commercial purposes)
+ * provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions, and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions, and the following disclaimer in the
+ *    documentation and/or materials provided with the distribution.
+ *
+ * 3. In addition, redistributions of modified forms of the source or binary
+ *    code must carry prominent notices stating that the original code was
+ *    changed and the date of the change.
+ *
+ *  4. All publications or advertising materials mentioning features or use of
+ *     this software are asked, but not required, to acknowledge that it was
+ *     developed by Intel Corporation and credit the contributors.
+ *
+ * 5. Neither the name of Intel Corporation, nor the name of any Contributor
+ *    may be used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER BE LIABLE FOR ANY
+ * DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
+ * THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * This is a simple example of cart rpc group test based on crt API.
+ */
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <stdint.h>
+#include <assert.h>
+#include <getopt.h>
+#include <semaphore.h>
+
+#include "tests_common.h"
+#include "crt_fake_events.h"
+#include "test_group_rpc.h"
+#include "test_group_np_common.h"
+
+void
+test_run(void)
+{
+	crt_group_t	*grp = NULL;
+	d_rank_list_t	*rank_list = NULL;
+	d_rank_t	 rank;
+	int		 tag;
+	crt_endpoint_t	 server_ep = {0};
+	crt_rpc_t	*rpc_req = NULL;
+	int		 i;
+	int		 rc = 0;
+
+	tc_cli_start_basic(test_g.t_local_group_name,
+			   test_g.t_remote_group_name,
+			   &grp, &rank_list, &test_g.t_crt_ctx[0],
+			   &test_g.t_tid[0], test_g.t_srv_ctx_num);
+
+	rc = sem_init(&test_g.t_token_to_proceed, 0, 0);
+	D_ASSERTF(rc == 0, "sem_init() failed.\n");
+
+	/* register RPCs */
+	rc = crt_proto_register(&my_proto_fmt_test_group2);
+	D_ASSERTF(rc == 0, "crt_proto_register() failed. rc: %d\n",
+			rc);
+
+	sleep(2);
```

Comment:
i wonder if we should just embed sleep(2) into tc_wait_for_ranks() instead now -- this way once we fix outstanding issues we will only need to remove sleep from a single location